src/iden3comm/handlers/fetch.ts

```diff
@@ -155,4 +183,98 @@ export class FetchHandler implements IFetchHandler {
     }
     return credentials;
   }
+
+  /**
+   * @inheritdoc IFetchHandler#handleCredentialFetchRequest
+   */
+  async handleCredentialFetchRequest(basicMessage: Uint8Array): Promise<Uint8Array> {
+    const msgRequest = await FetchHandler.unpackMessage<CredentialFetchRequestMessage>(
+      this._packerMgr,
+      basicMessage,
+      PROTOCOL_MESSAGE_TYPE.CREDENTIAL_FETCH_REQUEST_MESSAGE_TYPE
+    );
+
+    if (!msgRequest.to) {
+      throw new Error("failed request. empty 'to' field");
+    }
+
+    if (!msgRequest.from) {
+      throw new Error("failed request. empty 'from' field");
+    }
+
+    const issuerDID = DID.parse(msgRequest.to);
+    const userDID = DID.parse(msgRequest.from);
+
+    const credId = msgRequest.body?.id;
+    if (!credId) {
+      throw new Error('invalid claim id in fetch request body');
+    }
+
+    if (!this.opts?.credentialWallet) {
+      throw new Error('please provide credential wallet in options');
+    }
+
+    const cred = await this.opts.credentialWallet.findById(credId);
+
+    if (!cred) {
+      throw new Error('credential not found');
+    }
+
+    const userToVerifyDID = getUserDIDFromCredential(issuerDID, cred);
+
+    if (userToVerifyDID.string() !== userDID.string()) {
+      throw new Error('credential subject is not a sender DID');
+    }
+
+    return byteEncoder.encode(
+      JSON.stringify({
+        id: uuid.v4(),
+        type: PROTOCOL_MESSAGE_TYPE.CREDENTIAL_ISSUANCE_RESPONSE_MESSAGE_TYPE,
+        typ: msgRequest.typ ?? MediaType.PlainMessage,
```

Comment:
this is not right, You need to call package manager .pack() with MediaType.PlainMessage